c-api-examples/streaming-zipformer-buffered-tokens-hotwords-c-api.c

```diff
@@ -0,0 +1,197 @@
+// c-api-examples/streaming-zipformer-buffered-tokens-hotwords-c-api.c
+//
+// Copyright (c)  2024  Xiaomi Corporation
+// Copyright (c)  2024  Luo Xiao
+
+//
+// This file demonstrates how to use streaming Zipformer with sherpa-onnx's C
+// and with tokens and hotwords loaded from buffered strings instread external
+// files API.
+// clang-format off
+// 
+// wget https://github.com/k2-fsa/sherpa-onnx/releases/download/asr-models/sherpa-onnx-streaming-zipformer-en-20M-2023-02-17.tar.bz2
+// tar xvf sherpa-onnx-streaming-zipformer-en-20M-2023-02-17.tar.bz2
+// rm sherpa-onnx-streaming-zipformer-en-20M-2023-02-17.tar.bz2
+//
+// clang-format on
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "sherpa-onnx/c-api/c-api.h"
+
+size_t read_file(const char *filename, const char **buffer_out) {
+  FILE *file = fopen(filename, "rb");
+  if (file == NULL) {
+    fprintf(stderr, "Failed to open %s\n", filename);
+    return -1;
+  }
+  fseek(file, 0L, SEEK_END);
+  long size = ftell(file);
+  rewind(file);
+  *buffer_out = malloc(size + 1);
```

Comment:
```suggestion
  *buffer_out = malloc(size);
```
or to use
```c
(*buffer_out)[size] = '\0';
```
Otherwise, the last byte is not initialized.